shared/src/api/extension/api/configuration.test.ts

```diff
@@ -1,55 +1,101 @@
-import * as sinon from 'sinon'
-import { ExtConfiguration } from './configuration'
-import { Remote, proxyMarker, createEndpoint, releaseProxy } from 'comlink'
-import { ClientConfigurationAPI } from '../../client/api/configuration'
-import { noop } from 'lodash'
-import { addProxyMethods } from '../../util'
+import { Remote } from 'comlink'
+import { CalledFromExtHost } from '../../contract'
+import { initNewExtensionAPI } from '../newExtensionApi'
+import { SettingsEdit } from '../../client/services/settings'
 
-describe('ExtConfiguration', () => {
-    const PROXY: Remote<ClientConfigurationAPI> = addProxyMethods({
-        [proxyMarker]: Promise.resolve(true),
-        $acceptConfigurationUpdate: Object.assign(() => Promise.resolve(), {
-            [createEndpoint]: () => Promise.resolve(new MessagePort()),
-            [releaseProxy]: noop,
-        }),
-    })
+// Promisify method calls and objects if specified, throws otherwise
+const pretendRemote = (api: Partial<CalledFromExtHost>): Remote<CalledFromExtHost> =>
+    (new Proxy(api, {
+        get: (a, prop) => {
+            if (prop in a) {
+                if (typeof (a as any)[prop] !== 'function') {
+                    return Promise.resolve((a as any)[prop])
+                }
+
+                return (...args: any[]) => Promise.resolve((a as any)[prop](...args))
+            }
+            throw new Error(`unspecified property in the stub ${prop.toString()}`)
```

Comment:
I think this should return `undefined` to support optional properties and match JS semantics
```suggestion
            return undefined
```